components/os/unix.go

```diff
@@ -33,6 +37,10 @@ func (*Unix) GetRunAgentCmd(parameters string) string {
 	return "sudo datadog-agent " + parameters
 }
 
+func getAgentMajorVersion(version AgentVersion) string {
+	return string(version.RepoBranch[len(version.RepoBranch)-1])
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
This function works only if `version.RepoBranch` is not empty. It is also used when version is custom. What if you parsed the major version once in `parseVersion` in `params.go`, get rid of this function and use `agent.MajorVersion` in `GetInstallCmd` ?

Reply:
mm not sure I fully understood this part, this function will be called only when the IsCustomImage will be true, which will make sure that the repo branch is not empty. In the start I do added a variable that is holding only the major version, but then I have noticed that for example for pipeline-16370943-a7 it will include only the major - 7, I am not sure how useful it will be to test major 6. But lets discuss about it on private, if you think that the agent version should always be 7 I may remove this part and validate in early stages that it exists.